File: basic/src/main/java/top/non/late/pavilion/basic/java/designs/builder/code/BookBuilder.java
package top.non.late.pavilion.basic.java.designs.builder.code;

/**
 * Created by nonlatepavilion@outlook.com on 2017/9/7.
 * Desc: 一般用接口定义创建一个product所需要的各个操作<br/>
 * 该示例中，用于指定book的bookName，author，isbn
 */
public interface BookBuilder {

    void bookName();

    void author();

    void isbn();

    BookProduct build();

}

File: basic/src/main/java/top/non/late/pavilion/basic/kotlin/designs/builder/BookBuilder.kt
package top.non.late.pavilion.basic.kotlin.designs.builder

/**
 * Created by yuhp@terminus.io on 2017/9/14.
 * Desc:
 */
interface BookBuilder {
    fun bookName()
    fun author()
    fun isbn()
    fun build(): BookProduct
}